### PR TITLE
Accessibility fixes

### PR DESCRIPTION
**Pull request by helenxuyang**
### Summary 

This pull request is the first step towards improving accessibility

Home page:
- [x] fixed header on homepage to be fully accessible with a screen reader (before, explore by touch wouldn't sense the header)
- [x] replaced pull to refresh with refresh button
- [x] fixed semantics for current ride case (before, it was merging too many widgets and wasn't reading the call driver button)
- [x] added tap animations for ride cards 
- [x] removed sun from header :( and changed it to resize to fit in the available space next to the buttons

Ride page:
- [x] added alt text and button text on ride page 
- [x] fixed ScheduleBar semantics and added tap animation
- [x] added tap animation for location icons 
- [x] made location sheet responsive for text length 

Ride flow:
- [x] fixed validation error interactions and semantics (before, it would keep the error even after it was fixed, and errors weren't reading correctly after interaction)
- [x] fixed edit box tap interaction (before, didn't work with Switch Access)

Profile:
- [x] changed phone number semantics to read digit by digit (before, it would just read like "one billion" 
- [x] fixed arrow icon buttons on profile for tap area and animation  
- [x] increased tap area size for edit profile picture button
- [x] implemented terms of service and its semantics  

Onboarding:
- [x] pulled out widgets for reusability and so it'd be easier to implement accessibility
- [x] fixed car svg (was cut off on the top/right before) and Cornell svg (we were using the png before because the svg file was broken)
- [x] removed semantics for decorative images, added for text
- [x] implemented terms of service and its semantics  

General:
- [x] added responsive text scaling for RichText (it isn't automatically enabled)
- [x] fixed text overflows 

### Test Plan 
Tested with TalkBack, Switch Access (as much as I could with just next+select or next+scroll because I could only use 2 switches), and zoomed in text
Asked Laura and Tony to check with VoiceOver, tested as much as I could with Switch Access

### Notes 
- unsure if we need autofocus on every new page--also not quite sure if this is possible without a hacky solution

### CODE REVIEW

File: lib/pages/RidePage.dart
Comment (tyang98) [inline lib/pages/RidePage.dart:479]: On the ride page the cancel ride button isn't announced as a button when using voiceover. Is that intended?

File: lib/pages/Profile.dart
Comment (tyang98) [inline lib/pages/Profile.dart:188]: When it's the first time using the app/login, this button goes to the home screen when pressed. If my fixes with navigation already addresses this, then this is not an issue.

**Review by tyang98 — COMMENTED** [review]
Awesome work, accessibility on rider is looking great! I won't approve right now because there are some things to be addressed. Additionally, I had some trouble creating a current ride last night so I have not checked over that page yet.

File: lib/pages/Terms.dart
Comment (helenxuyang) [reply lib/pages/Terms.dart:8]: Good catch, thanks! I made this page scrollable to be responsive for large text, so I'll change it so that the button is sticky at the bottom like we did for the ride flow pages

File: lib/pages/ride-flow/FlowWidgets.dart
Comment (helenxuyang) [reply lib/pages/ride-flow/FlowWidgets.dart:27]: Good catch, missed this after merging with the font change branch. Will bump up the font weight in CarriageTheme

File: lib/pages/RidePage.dart
Comment (helenxuyang) [reply lib/pages/RidePage.dart:479]: Good catch, thanks! Will wrap with a Semantics so that it reads properly

File: lib/pages/RidePage.dart
Comment (helenxuyang) [reply lib/pages/RidePage.dart]: Discussed offline - seems like VoiceOver doesn't do this even in other apps while TalkBack does, so we may not have to do this. If needed, we'll parse the last 5 characters of the address which should be the zip code and add it to the semanticsLabel

File: lib/pages/RidePage.dart
Comment (helenxuyang) [reply lib/pages/RidePage.dart]: Discussed offline - checking with designers to see if it's the whole card or just the button that should be tappable

File: lib/pages/Profile.dart
Comment (helenxuyang) [reply lib/pages/Profile.dart:189]: Fixed this up, thanks!

File: lib/pages/RidePage.dart
Comment (tyang98) [inline lib/pages/RidePage.dart:222]: Right now it's just saying Button, do we want "Cancel Ride, Button"?

**Review by tyang98 — APPROVED** [review]
Thanks for addressing all of my comments; everything looks good to go. There is one last question I have that might need to be addressed but I will approve for now and if you decide to make the change you can merge this PR after.

File: lib/pages/RidePage.dart
Comment (helenxuyang) [reply lib/pages/RidePage.dart:222]: That's strange, it reads "Cancel Ride, Button" with TalkBack. I'll move the button label out to the Semantics just to be safe

Comment (tyang98) [reply lib/pages/RidePage.dart:228]: It works now, thanks for changing

**Pull request merged**
